src/lib/gundb/UserStorage.js

```diff
@@ -404,22 +425,31 @@ class UserStorage {
    * need to develop for gundb immutable keys to non first user
    */
   async indexProfileField(field: string, value: string, privacy: FieldPrivacy): Promise<ACK> {
-    if (!UserStorage.indexableFields[field]) return
+    if (!UserStorage.indexableFields[field]) return Promise.resolve({ err: 'Not indexable field', ok: 0 })
     const cleanValue = UserStorage.cleanFieldForIndex(field, value)
-
+    const indexNode = gun.rootAO(`users/by${field}`).get(cleanValue)
     logger.info({ field, value, privacy })
-    if (privacy !== 'public')
-      return gun
-        .get('users')
-        .get('by' + field)
-        .get(cleanValue)
-        .putAck(null)
-
-    const gunResult = await gun
-      .get('users')
-      .get('by' + field)
-      .get(cleanValue)
-      .putAck(this.gunuser)
+
+    const indexValue = await indexNode.then()
+    logger.info({
+      field,
+      value,
+      privacy,
+      indexValue: indexValue,
+      currentUser: this.gunuser.is.pub
+    })
+
+    if (indexValue && indexValue.pub != this.gunuser.is.pub) {
+      return Promise.resolve({ err: `Existing index on field ${field}`, ok: 0 })
+    }
+
+    if (privacy !== 'public') {
+      const result = indexNode.putAck(null)
+      logger.info('Result: ', result)
+      return Promise.resolve({ err: 'Not public field', ok: 0 })
+    }
+
+    const gunResult = await indexNode.putAck(this.gunuser)
```

Comment:
`result` in line `447` pursues the same goal as this `gunResult`, maybe using the same name would make it easier to reason about.

what you think?

Reply:
@dminones This shouldn't be an error. if privacy is not public we set the field to null this is the expected behaviour.
This also requires changes to setProfileField, currently indexfield is called only if privacy is public, but maybe privacy was changed from public to private. so indexprofilefield should be called every time